google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/ReadOnlyStalenessUtil.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.jdbc;
+
+import com.google.api.client.util.DateTime;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.ErrorCode;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.TimestampBound;
+import com.google.cloud.spanner.TimestampBound.Mode;
+import com.google.common.base.Strings;
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Durations;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Util class for parsing and converting ReadOnlyStaleness values to/from strings. This util is used
+ * to parse client side statements and values for read only staleness for read-only transactions on
+ * Cloud Spanner.
+ */
+class ReadOnlyStalenessUtil {
+  private static final TimeZone GMT = TimeZone.getTimeZone("GMT");
+  /** Regular expression for parsing RFC3339 times with nanosecond precision. */
+  private static final String RFC3339_REGEX =
```

Comment:
There is an RFC parser in the http client. Can we please switch to that? https://github.com/googleapis/google-http-java-client/blob/master/google-http-client/src/main/java/com/google/api/client/util/DateTime.java

Reply:
The RFC parser in the http client has millisecond precision, both for parsing and converting to string values. Switching to that one is possible, but would mean that we would not be able to support microsecond or nanosecond precision.

Reply:
In that case, let's add the enhanced precision to the http client repository, and use the current http client implementation with a todo for now since this is still in a branch.

Reply:
I've changed the JDBC implementation to use the parser in the http client, and I've added a PR in the http client for adding nanosecond support.